pypeit/core/flexure.py

```diff
@@ -279,9 +246,12 @@ def spec_flex_shift(obj_skyspec, arx_skyspec, arx_lines, mxshft=20, excess_shft=
         fit.fit()
         max_fit = -0.5 * fit.fitc[1] / fit.fitc[2]
 
+        shift = float(max_fit) - lag0
         # Deal with the case of shifts greater than ``mxshft``
-        shift = float(max_fit)-lag0
-        if shift > mxshft:
+        # We need to compare the absolute value of shift to ``mxshft``, since shift can be
+        # positive or negative, while ``mxshft`` is generally only positive
+        # We use the floor of shift to avoid to trigger the error/warning for differences <1pixel
+        if np.floor(abs(shift)).astype(int) > mxshft:
```

Comment:
I used `floor` so that if `abs(shift)` is larger that  `mxshft` by 0.5 or even 0.9, the code does not dive an error. It's just to have this condition not too strict.
But I guess I can just use `int`, I don't need `floor`.